neurom/apps/morph_stats.py

```diff
@@ -112,7 +116,14 @@ def extract_dataframe(neurons, config):
     if 'neuron' in config:
         del config['neuron']
 
-    stats = {nrn.name: extract_stats(nrn, config) for nrn in neurons}
+    with multiprocessing.Pool(n_workers) as pool:
+        stats = {nrn if isinstance(nrn, str) else nrn.name:
+                 stat for nrn, stat in tqdm(zip(neurons,
+                                                pool.imap(partial(extract_stats, config=config),
+                                                          neurons)
+                                                ), total=len(neurons)
+                                            )
```

Comment:
Instead of changing `extract_stats` signature, you could do this

```suggestion
        def _compute(nrn):
            if not isinstance(nrn, Neuron):
                nrn = load_neuron(nrn)
            return nrn.name, extract_stats(nrn, config=config)
        stats = dict(tqdm(pool.imap(_compute, neurons)), total=len(neurons))
```
This will also fix the fact that passing a string in your case would return an incorrect neuron name since it would use the whole string instead of just the last part of the path.

Reply:
aaah I did the stupid thing of copy pasting and you messed up the brakets, so it was added a key total in the dict haha